### ISSUE TITLE
[BUG] SPDK json rpc server returns invalid json string

### ISSUE THREAD

**Issue by derekbit** [longhorn/longhorn#6190]
## Describe the bug (🐛 if you encounter this issue)

This is a weird json response from spdk json server.
```
{"jsonrpc":"2.0","id":7516,"result":[{"name":"disk-1","aliases":["5dbfed67-547b-4f27-acfb-5a398307d370"],"product_name":"AIO disk","block_size":4096,"num_blocks":13107200,"uuid":"5dbfed67-547b-4f27-acfb-5a398307d370","assigned_rate_limits":{"rw_ios_per_sec":0,"rw_mbytes_per_sec":0,"r_mbytes_per_sec":0,"w_mbytes_per_sec":0},"claimed":true,"claim_type":"read_many_write_one","zoned":false,"supported_io_types":{"read":true,"write":true,"unmap":false,"write_zeroes":true,"flush":true,"reset":true,"compare":false,"compare_and_write":false,"abort":false,"nvme_admin":false,"nvme_io":false},"driver_specific":{"aio":{"filename":"/host/dev/loop9","block_size_override":true,"readonly":false}}},{"name":"f47eefa1-231a-4467-b268-aeae33e9bfd6","aliases":["disk-1/pvc-132d5906-3bda-405e-bd38-8094a2435151-r-f21516f3"],"product_name":"Logical Volume","block_size":4096,"num_blocks":5242880,"uuid":"f47eefa1-231a-4467-b268-aeae33e9bfd6","assigned_rate_limits":{"rw_ios_per_sec":0,"rw_mbytes_per_sec":0,"r_mbytes_per_sec":0,"w_mbytes_per_sec":0},"claimed":true,"claim_type":"exclusive_write","zoned":false,"supported_io_types":{"read":true,"write":true,"unmap":true,"write_zeroes":true,"flush":false,"reset":true,"compare":false,"compare_and_write":false,"abort":false,"nvme_admin":false,"nvme_io":false},"driver_specific":{"lvol":{"lvol_store_uuid":"f030b862-d4fa-421a-86ba-becb465dcb61","base_bdev":"disk-1","thin_provision":true,"snapshot":false,"clone":true,"base_snapshot":"pvc-132d5906-3bda-405e-bd38-8094a2435151-r-f21516f3-snap-rebuild-5e43a0c7","esnap_clone":false}}},{"name":"f798d6b5-2596-40bb-9b59-e3922e103cbf","aliases":["disk-1/pvc-132d5906-3bda-405e-bd38-8094a2435151-r-f21516f3-snap-rebuild-52db2645"],"product_name":"Logical Volume","block_size":4096,"num_blocks":5242880,"uuid":"f798d6b5-2596-40bb-9b59-e3922e103cbf","assigned_rate_limits":{"rw_ios_per_sec":0,"rw_mbytes_per_sec":0,"r_mbytes_per_sec":0,"w_mbytes_per_sec":0},"claimed":false,"zoned":false,"supported_io_types":{"read":true,"write":false,"unmap":false,"write_zeroes":false,"flush":false,"reset":true,"compare":false,"compare_and_write":false,"abort":false,"nvme_admin":false,"nvme_io":false},"driver_specific":{"lvol":{"lvol_store_uuid":"f030b862-d4fa-421a-86ba-becb465dcb61","base_bdev":"disk-1","thin_provision":true,"snapshot":true,"clone":false,"clones":["pvc-132d5906-3bda-405e-bd38-8094a2435151-r-f21516f3-snap-rebuild-5e43a0c7"],"esnap_clone":false}}},{"name":"2d2ae716-8968-4f64-99e3-b3a1dd638718","aliases":["disk-1/pvc-132d5906-3bda-405e-bd38-8094a2435151-r-f21516f3-snap-rebuild-5e43a0c7"],"product_name":"Logical Volume","block_size":4096,"num_blocks":5242880,"uuid":"2d2ae716-8968-4f64-99e3-b3a1dd638718","assigned_rate_limits":{"rw_ios_per_sec":0,"rw_mbytes_per_sec":0,"r_mbytes_per_sec":0,"w_mbytes_per_sec":0},"claimed":false,"zoned":false,"supported_io_types":{"read":true,"write":false,"unmap":false,"write_zeroes":false,"flush":false,"reset":true,"compare":false,"compare_and_write":false,"abort":false,"nvme_admin":false,"nvme_io":false},"driver_specific":{"lvol":{"lvol_store_uuid":"f030b862-d4fa-421a-86ba-becb465dcb61","base_bdev":"disk-1","thin_provision":true,"snapshot":true,"clone":true,"base_snapshot":"pvc-132d5906-3bda-405e-bd38-8094a2435151-r-f21516f3-snap-rebuild-52db2645","clones":["pvc-132d5906-3bda-405e-bd38-8094a2435151-r-f21516f3"],"esnap_clone":false}}},{"name":"pvc-132d5906-3bda-405e-bd38-8094a2435151-r-ce75dfbfn1","aliases":["0420c293-c74a-4867-a5f2-65300e30b81d"],"product_name":"NVMe disk","block_size":4096,"num_blocks":5242880,"uuid":"0420c293-c74a-4867-a5f2-65300e30b81d","assigned_rate_limits":{"rw_ios_per_sec":0,"rw_mbytes_per_sec":0,"r_mbytes_per_sec":0,"w_mbytes_per_sec":0},"claimed":true,"claim_type":"exclusive_write","zoned":false,"supported_io_types":{"read":true,"write":true,"unmap":true,"write_zeroes":true,"flush":true,"reset":true,"compare":true,"compare_and_write":true,"abort":true,"nvme_admin":true,"nvme_io":true},"driver_specific":{"nvme":[{"trid":{"trtype":"TCP","adrfam":"IPv4","traddr":"10.42.216.170","trsvcid":"20001","subnqn":"nqn.2023-01.io.longhorn.spdk:pvc-132d5906-3bda-405e-bd38-8094a2435151-r-ce75dfbf"},"ctrlr_data":{"cntlid":1,"vendor_id":"0x8086","model_number":","serial_number":"00000000000000000000","firmware_revision":"23.05","subnqn":"nqn.2023-01.io.longhorn.spdk:pvc-132d5906-3bda-405e-bd38-8094a2435151-r-ce75dfbf","oacs":{"security":0,"format":0,"firmware":0,"ns_manage":0},"multi_ctrlr":true,"ana_reporting":false},"vs":{"nvme_version":"1.3"},"ns_data":{"id":1,"can_share":true}}],"mp_policy":"active_passive"}},{"name":"pvc-132d5906-3bda-405e-bd38-8094a2435151-r-7716870bn1","aliases":["e05c1400-4c36-4ed0-a287-03f928b73aef"],"product_name":"NVMe disk","block_size":4096,"num_blocks":5242880,"uuid":"e05c1400-4c36-4ed0-a287-03f928b73aef","assigned_rate_limits":{"rw_ios_per_sec":0,"rw_mbytes_per_sec":0,"r_mbytes_per_sec":0,"w_mbytes_per_sec":0},"claimed":true,"claim_type":"exclusive_write","zoned":false,"supported_io_types":{"read":true,"write":true,"unmap":true,"write_zeroes":true,"flush":true,"reset":true,"compare":true,"compare_and_write":true,"abort":true,"nvme_admin":true,"nvme_io":true},"driver_specific":{"nvme":[{"trid":{"trtype":"TCP","adrfam":"IPv4","traddr":"10.42.84.212","trsvcid":"20001","subnqn":"nqn.2023-01.io.longhorn.spdk:pvc-132d5906-3bda-405e-bd38-8094a2435151-r-7716870b"},"ctrlr_data":{"cntlid":1,"vendor_id":"0x8086","model_number":"SPDK bdev Controller","serial_number":"00000000000000000000","firmware_revision":"23.05","subnqn":"nqn.2023-01.io.longhorn.spdk:pvc-132d5906-3bda-405e-bd38-8094a2435151-r-7716870b","oacs":{"security":0,"format":0,"firmware":0,"ns_manage":0},"multi_ctrlr":true,"ana_reporting":false},"vs":{"nvme_version":"1.3"},"ns_data":{"id":1,"can_share":true}}],"mp_policy":"active_passive"}},{"name":"pvc-132d5906-3bda-405e-bd38-8094a2435151-e-6f65e425","aliases":["7dff52e1-1502-4d14-9bb2-acf41d097808"],"product_name":"Raid Volume","block_size":4096,"num_blocks":5242880,"uuid":"7dff52e1-1502-4d14-9bb2-acf41d097808","assigned_rate_limits":{"rw_ios_per_sec":0,"rw_mbytes_per_sec":0,"r_mbytes_per_sec":0,"w_mbytes_per_sec":0},"claimed":true,"claim_type":"exclusive_write","zoned":false,"supported_io_types":{"read":true,"write":true,"unmap":false,"write_zeroes":true,"flush":false,"reset":true,"compare":false,"compare_and_write":false,"abort":false,"nvme_admin":false,"nvme_io":false},"driver_specific":{"raid":{"strip_size_kb":0,"state":"online","raid_level":"raid1","superblock":false,"num_base_bdevs":3,"num_base_bdevs_discovered":3,"num_base_bdevs_operational":3,"base_bdevs_list":[{"name":"pvc-132d5906-3bda-405e-bd38-8094a2435151-r-ce75dfbfn1","uuid":"0420c293-c74a-4867-a5f2-65300e30b81d","is_configured":true,"data_offset":0,"data_size":5242880},{"name":"disk-1/pvc-132d5906-3bda-405e-bd38-8094a2435151-r-f21516f3","uuid":"f47eefa1-231a-4467-b268-aeae33e9bfd6","is_configured":true,"data_offset":0,"data_size":5242880},{"name":"pvc-132d5906-3bda-405e-bd38-8094a2435151-r-7716870bn1","uuid":"e05c1400-4c36-4ed0-a287-03f928b73aef","is_configured":true,"data_offset":0,"data_size":5242880}]}}}]}
```

The problematic field is `model_number: ",`. A double quote char is missing, so longhorn-instance-manager emits 
```
[longhorn-instance-manager] time="2023-06-25T15:34:20Z" level=error msg="SPDK Server: failed to verify and update replica cache, will retry later" error="error sending message, id 18737, method bdev_lvol_get_lvstores, params { }: invalid character 's' after object key:value pair"
```

cc @DamiaSan @shuo-wu @innobead 

## To Reproduce

3. Perform '....'

## Log or Support bundle

If applicable, add the Longhorn managers' log or support bundle when the issue happens. 
You can generate a Support Bundle using the link at the footer of the Longhorn UI.

## Environment

 - Longhorn version:
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl):
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version:
   - Number of management node in the cluster:
   - Number of worker node in the cluster:
 - Node config
   - OS type and version:
   - CPU per node:
   - Memory per node:
   - Disk type(e.g. SSD/NVMe):
   - Network bandwidth between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal):
 - Number of Longhorn volumes in the cluster:

**Comment by innobead** [comment]
@derekbit is this randomly happening?

**Comment by derekbit** [comment]
> is this randomly happening? 

Yes.

> will this get back to normal eventually?

Need to improve the error handling in instance-manager pod when running into the error

**Comment by DamiaSan** [comment]
Maybe it is not a missing `"` but it is an empty string enclosed by single quote characters. Is `model_number` always empty? If not, when it is valorized, is it enclosed by single or double quotes?

**Comment by derekbit** [comment]
No, it is a double quote char.

**Comment by derekbit** [comment]
We don't set the value.

**Comment by DamiaSan** [comment]
Forcing the debug I could reproduce the issue:
```
{"jsonrpc":"2.0","id":1,"result":[{"nqn":"nqn.2014-08.org.nvmexpress.discovery","subtype":"Discovery","listen_addresses":[],"allow_any_host":true,"hosts":[]},{"nqn":"nqn.2023-01.io.longhorn.spdk:pvc-132d5906-3bda-405e-bd38-8094a2435151-r-ce75dfbf","subtype":"NVMe","listen_addresses":[],"allow_any_host":true,"hosts":[],"serial_number":"00000000000000000000","model_number":","max_namespaces":32,"min_cntlid":1,"max_cntlid":65519,"namespaces":[]}]}
```
This behaviour happens when the decoder doesn't recognize a valid UTF8 character: https://github.com/longhorn/spdk/blob/575a9a8669b7b9a4c73b42af6f51b2c2ab54c7b4/lib/json/json_write.c#L418.
But I don't understand how it can happen ...

**Comment by derekbit** [comment]
> Forcing the debug I could reproduce the issue:

How did you reproduce it?

**Comment by DamiaSan** [comment]
With debugger, I set `codepoint_len` to a bad value (5 for example) and so the function returns with `fail(w);`

**Comment by derekbit** [comment]
Got it. I think running into the same result is expected by your method.
Currently, I'm not sure it is caused by the `fail(w)` or other trim operations.

**Comment by innobead** [comment]
@derekbit is this still an issue? I think we already fixed it in our forked version. If yes, let's close this issue.

**Comment by derekbit** [comment]
> @derekbit is this still an issue? I think we already fixed it in our forked version, also in 1.5.0. If yes, let's close this issue.

Nope. It is resolved. We can close it.
